controllers/operatorpolicy_controller.go

```diff
@@ -791,16 +974,32 @@ func (r *OperatorPolicyReconciler) handleInstallPlan(
 
 	// InstallPlans are generally kept in order to provide a history of actions on the cluster, but
 	// they can be deleted without impacting the installed operator. So, not finding any should not
-	// be considered a reason for NonCompliance.
+	// be considered a reason for NonCompliance, regardless of musthave or mustnothave.
 	if len(ownedInstallPlans) == 0 {
-		return updateStatus(policy, noInstallPlansCond, noInstallPlansObj(sub.Namespace)), nil
+		return nil, updateStatus(policy, noInstallPlansCond, noInstallPlansObj(sub.Namespace)), nil
 	}
 
+	if policy.Spec.ComplianceType.IsMustHave() {
+		changed, err := r.musthaveInstallPlan(ctx, policy, sub, ownedInstallPlans)
+
+		return nil, changed, err
+	}
+
+	return r.mustnothaveInstallPlan(ctx, policy, ownedInstallPlans)
+}
+
+func (r *OperatorPolicyReconciler) musthaveInstallPlan(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	sub *operatorv1alpha1.Subscription,
+	ownedInstallPlans []unstructured.Unstructured,
+) (bool, error) {
 	OpLog := ctrl.LoggerFrom(ctx)
 	relatedInstallPlans := make([]policyv1.RelatedObject, len(ownedInstallPlans))
 	ipsRequiringApproval := make([]unstructured.Unstructured, 0)
```

Comment:
```suggestion
	relatedInstallPlans := make([]policyv1.RelatedObject, 0, len(ownedInstallPlans))
	ipsRequiringApproval := make([]unstructured.Unstructured)
```

Reply:
I've now done a search on `make([]` to find these instances 😆 thanks for noticing them.